Dockerfile

```diff
@@ -19,26 +20,27 @@ FROM python:3.8-slim-bullseye
 SHELL ["/bin/bash", "-c"]
 COPY --from=builder /usr/local/lib/python3.8 /usr/local/lib/python3.8
 
+ARG optional_dependencies=dev,voikko,pycld3,fasttext,nn,omikuji,yake,spacy
 # Install system dependencies needed at runtime:
-RUN apt-get update \
-	&& apt-get install -y --no-install-recommends \
-		libvoikko1 \
-		voikko-fi \
-	# For Docker healthcheck:
-	&& apt-get install -y --no-install-recommends curl \
-	&& rm -rf /var/lib/apt/lists/* /usr/include/*
+RUN apt-get update && \
+	if [[ $optional_dependencies =~ "voikko" ]]; then \
+		apt-get install -y --no-install-recommends \
+			libvoikko1 \
+			voikko-fi; \
+	fi && \
+	# curl for Docker healthcheck and rsync for model transfers:
+	apt-get install -y --no-install-recommends curl rsync && \
+	rm -rf /var/lib/apt/lists/* /usr/include/*
 
 WORKDIR /Annif
-RUN pip install --upgrade pip --no-cache-dir
+RUN pip install --upgrade pip setuptools --no-cache-dir
```

Comment:
Or maybe this would be a better place for `wheel`...

Reply:
And here too.